### ISSUE TITLE
Multiple Test Notifications

### ISSUE THREAD

**Issue by Unthrottled** [zd-zero/waifu-motivator-plugin#124]
**Describe the bug**
The test pass notification is invoked more than once on completion, causing the plugin to dispatch more than one waifu to tell you that you did a good job. 

**To Reproduce**
Run this test: https://github.com/Unthrottled/hdd/blob/master/src/test/java/io/acari/water/WaterRepositoryTest.java

**Expected behavior**
Well you get one sound played, but you get multiple notifications the (1 more) at the bottom of the window

**Screenshots**
This is the event log and the notification after one test run.

![image](https://user-images.githubusercontent.com/15972415/88380157-a1eac800-cd69-11ea-9333-db5075733316.png)

**Platform (please complete the following information):**
 - Version WMP 1.0
```
IntelliJ IDEA 2020.1 (Community Edition)
Build #IC-201.6668.121, built on April 8, 2020
Runtime version: 11.0.6+8-b765.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.15.0-109-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 512M
Cores: 12
Non-Bundled Plugins: io.acari.DDLCTheme, zd.zero.waifu-motivator-plugin
Current Desktop: ubuntu:GNOME
```

**Additional context**
Only one listener is being set, there is just more than one test pass notification being sent to the listeners.

**Comment by Unthrottled** [comment]
I can look into the fix.

**Comment by zd-zero** [comment]
I also had experienced this in #110 but it only happens when there are multiple projects opened, I can only reproduce this issue with the one you've [provided](https://github.com/Unthrottled/hdd/blob/master/src/test/java/io/acari/water/WaterRepositoryTest.java). Do you have any idea why it shows up there and not on the other tests?

My initial investigation, I tried to recreate it in a different project
### Can't reproduce with
1. Parameterized Test
<img width="1409" alt="Screen Shot 2020-07-25 at 04 43 39" src="https://user-images.githubusercontent.com/21978370/88434263-69fb8780-ce32-11ea-9dd1-4fe2aad1d7e0.png">

2. `n` Number of individual test
<img width="1419" alt="Screen Shot 2020-07-25 at 04 48 28" src="https://user-images.githubusercontent.com/21978370/88434288-77b10d00-ce32-11ea-8272-b65a26a28de4.png">

### Can reproduce with [this test](https://github.com/Unthrottled/hdd/blob/master/src/test/java/io/acari/water/WaterRepositoryTest.java). 
<img width="1414" alt="Screen Shot 2020-07-25 at 04 49 07" src="https://user-images.githubusercontent.com/21978370/88434401-aa5b0580-ce32-11ea-97e9-c4a858103775.png">

So turns out that there are different ways IntelliJ executes a test. As you would notice from the highlights, there are hierarchies of subtest or list of tests where I can't reproduce the issue, but to the one where I can reproduce it only shows one list of test but runs multiple tests.

_Note:_ I just got curious why this happens 😂

I'll check the PR #125 tomorrow as I was caught up by the rabbit hole when investigating this 😅. 

Me:
_When you've thought that you've resolved the issue of multiple test invocations._

Jetbrains unit test event:
![there-is-another](https://user-images.githubusercontent.com/21978370/88433582-f311bf00-ce30-11ea-9f22-efd91d99cb50.jpg)

**Comment by Unthrottled** [comment]
If you where curious here is what I found:

Both of these classes:

- https://github.com/JetBrains/intellij-community/blob/master/platform/smRunner/src/com/intellij/execution/testframework/sm/runner/GeneralTestEventsProcessor.java#L220
- https://github.com/JetBrains/intellij-community/blob/master/plugins/gradle/java/src/execution/test/runner/GradleTestsExecutionConsoleManager.java#L142

Both execute the notification topic publishing code. Which is consistent with the double notifications that I was experiencing. 

I looked into wiring our own listener, but I did not try to hard because I came up with a hack that fixes it, so I could work on more interesting problems.

![Disregard perfection, acquire fixes](https://i.imgflip.com/1csin4.jpg)

**Comment by zd-zero** [comment]
Woah, that is indeed a good find! Now that makes sense. 

> I looked into wiring our own listener, but I did not try too hard because I came up with a hack that fixes it, so I could work on more interesting motivator problems.

I think that it was the simplest implementation we can do to prevent multiple notifications, it was good ✨

**Comment by zd-zero** [comment]
Resolved by #125
